src/SuperVault.sol

```diff
@@ -215,42 +220,46 @@ contract SuperVault is BaseStrategy, ISuperVault {
 
     /// @inheritdoc ISuperVault
     function rebalance(RebalanceArgs calldata rebalanceArgs_) external payable override onlySuperVaultsStrategist {
-        uint256 lenRebalanceFrom = rebalanceArgs_.superformIdsRebalanceFrom.length;
-        uint256 lenAmountsRebalanceFrom = rebalanceArgs_.amountsRebalanceFrom.length;
-        uint256 lenFinal = rebalanceArgs_.finalSuperformIds.length;
+        TransientContext.set(bytes32(0), rebalanceArgs_.superformIdsRebalanceFrom.length); // lenRebalanceFrom
+        TransientContext.set("0x1", rebalanceArgs_.amountsRebalanceFrom.length); // lenAmountsRebalanceFrom
+        TransientContext.set("0x2", rebalanceArgs_.finalSuperformIds.length); // lenFinal
 
-        if (lenAmountsRebalanceFrom == 0) revert EMPTY_AMOUNTS_REBALANCE_FROM();
-        if (lenFinal == 0) revert EMPTY_FINAL_SUPERFORM_IDS();
+        if (TransientContext.get("0x1") == 0) revert EMPTY_AMOUNTS_REBALANCE_FROM();
```

Comment:
transient is cheaper than storage but still more exspensive than accessing calldata or memory (100 gas)

so I think the 0 checks can be beter done with the original calldata based values